keps/sig-cluster-lifecycle/kubeadm/4656-add-kubelet-instance-configuration/README.md

```diff
@@ -0,0 +1,287 @@
+## Design Details
+
+**Alpha(v1.31):** 
+
+kubeadm init:
+
+* No longer need to write the `--container-runtime-endpoint` to `/var/lib/kubelet/kubeadm-flags.env`.
+* No longer need to add the `kubeadm.alpha.kubernetes.io/cri-socket` annotation.
+* If the CRI socket provided in the kubeadm configuration is set, it is used first and the `/var/lib/kubelet/instace-config.yaml` configuration file is generated based on it. If the CRI socket is not set, the container runtime endpoint is automatically detected uploaded to the global configuration, and `/var/lib/kubelet/instace-config.yaml` will be generated.
+
+kubeadm join:
+
+* No longer need to add the `kubeadm.alpha.kubernetes.io/cri-socket` annotation.
+* If the CRI socket provided in the kubeadm configuration is set, it will take precedence and generate the `/var/lib/kubelet/instace-config.yaml` configuration file based on it, overwriting the kubelet configuration downloaded from the global configuration; if the CRI socket is not set, the kubelet configuration downloaded from the global configuration is used, and `/var/lib/kubelet/instace-config.yaml` is generated based on it.
+
+kubeadm reset:
+
+* There is no need to do anything, as `/var/lib/kubelet/instace-config.yaml` will be cleaned up with the deletion of `/var/lib/kubelet.`
+
+kubeadm upgrade apply:
+
+* `kubeadm upgrade apply` will check the `--container-runtime-endpoint` args in the `/var/lib/kubelet/kubeadm-flags.env` file and generate `/var/lib/kubelet/instace-config.yaml`, and override the `ContainerRuntimeEndpoint` field to `/var/lib/kubelet/config.yaml`.
+
+kubeadm upgrade node:
+
+* for other nodes, `kubeadm upgrade node` will check the `--container-runtime-endpoint` args in the `/var/lib/kubelet/kubeadm-flags.env` file and generate `/var/lib/kubelet/instace-config.yaml` based on them, and override the `ContainerRuntimeEndpoint` field to `/var/lib/kubelet/config.yaml`.
+
+**Beta(v1.32)**:
+
+* `kubeadm upgrade apply/node`  will use `/var/lib/kubelet/instace-config.yaml`, and override the `ContainerRuntimeEndpoint` field to `/var/lib/kubelet/config.yaml`.
+
+**GA(v1.34)**:
+
+* `kubeadm upgrade apply/node` will use `/var/lib/kubelet/instace-config.yaml` and remove CRI annotations from Node object and remove `--container-runtime-endpoint` args in `/var/lib/kubelet/kubeadm-flags.env` file
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+At least the following kubeadm packages will require updates and new unit tests:
+
+* `cmd/kubeadm/app/cmd/phases/init`
+
+* `cmd/kubeadm/app/phases/kubelet`
+
+* `cmd/kubeadm/app/phases/upgrade`
+
+* `cmd/kubeadm/app/cmd/phases/join`
+
+##### Integration tests
+
+- N/A
+
+##### e2e tests
+
+* A new e2e test will be added by using the kinder tool.
+
+### Graduation Criteria
```

Comment:
so there is a problem with the KEP currently

feature gates have two states: TRUE and FALSE

but the "Design details" and "Graduation Criteria" sections are organized as we both using and not using a feature gate, which is confusing.

your doc should be organized like this:

- in "Design details" explain the new feature gate
- say in which releases it will be alpha, beta, GA
- explain when it will be enabled by default
- explain what happens when it's enabled and not enabled to the different kubeadm commands

Reply:
I made some changes, please check if this is what we expected.